[Bug Report] RO creation fails, when buckets differ.

### Overview of the Issue

When creating an RO instance, if the bucket parameter `initialize_objectstore_bucket` of the source instance is different from the `objectstore_bucket` parameter of the RO instance, the creation will fail.

### Reproduction Steps

```
mysqld \
        --defaults-file=/u01/my_ro.cnf \
        --initialize-insecure \
        --initialize_from_objectstore=true \
        --initialize_smartengine_objectstore_data=true \
        --initialize_objectstore_provider=minio \
        --initialize_objectstore_region= \
        --initialize_objectstore_endpoint=http://172.17.0.2:9000 \
        --initialize_objectstore_bucket=wesql-storage \
        --initialize_repo_objectstore_id=sysbench \
        --objectstore_provider=minio \
        --objectstore_endpoint=http://172.17.0.2:9000 \
        --objectstore_region= \
        --objectstore_bucket=wesql-ro \
        --repo_objectstore_id=sysbench \
        --raft-replication-force-change-meta=ON \
        --core-file \
        --gdb
```

### Binary Version

```sh
apecloud/wesql-server:8.0.35-0.1.0_beta3.38
```

### Operating System and Environment details

```sh
all
```

### Log Fragments

```sh
2024-12-05T08:18:10.628024Z 0 [System] [MY-025144] [Server] Snapshot recovery: Initialize database from source object store snapshot  provider=minio region= endpoint=http://172.17.0.2:9000 bucket=wesql-storage repo_objectsotre_id=sysbench branch_objectsotre_id=main to provider=minio region= endpoint=http://172.17.0.2:9000 bucket=wesql-ro repo_objectsotre_id=sysbench branch_objectsotre_id=main
2024-12-05T08:18:10.630309Z 0 [System] [MY-025144] [Server] Snapshot recovery: Initialize smartengine object store data 
2024-12-05T08:18:10.635549Z 0 [System] [MY-025144] [Server] Snapshot recovery: write snapshot recovery status file.
2024-12-05T08:18:10.639165Z 0 [Note] [MY-025144] [Server] Snapshot recovery: list persistent objects sysbench/main/snapshot/snapshot.
2024-12-05T08:18:10.639169Z 0 [System] [MY-025144] [Server] Snapshot recovery: fetch last persistent snapshot index, no persistent snapshot index file
2024-12-05T08:18:10.639174Z 0 [ERROR] [MY-025144] [Server] Snapshot recovery: Failed to fetch last persistent snapshot index file.
```

--- a/mysql-test/suite/smartengine_consistent_snapshot/t/consistent_snapshot_ro_clone.test
+++ b/mysql-test/suite/smartengine_consistent_snapshot/t/consistent_snapshot_ro_clone.test
@@ -40,10 +40,11 @@ let $tmp_dir = $MYSQLTEST_VARDIR/tmp;
 let MYSQLD_DATADIR = $tmp_dir/snapshot_clone_ro_test;
 let MYSQLD_ERROR_LOG = $tmp_dir/ro_boost.err;
 let MYSQLD_ERROR_START_LOG = $tmp_dir/ro_start.err;
-let MYSQLD_EXTRA_INIT_ARGS = --initialize_from_objectstore=true --initialize_smartengine_objectstore_data=true --initialize_objectstore_region=$source_region --initialize_objectstore_bucket=$source_bucket --initialize_repo_objectstore_id=$source_repo;
+let self_bucket=wesql-ro;
+let MYSQLD_EXTRA_INIT_ARGS = --initialize_from_objectstore=true --initialize_smartengine_objectstore_data=true --initialize_objectstore_region=$source_region --initialize_objectstore_bucket=$source_bucket --initialize_repo_objectstore_id=$source_repo --objectstore_bucket=$self_bucket;
 
 --echo # Initialize ro server
-let $MYSQLD_EXTRA_ARGS = --port=$ro_port --socket=$MYSQLD_DATADIR/test.sock --console --skip-ssl --log-bin=master-bin --objectstore-region=.local_region --log-error-verbosity=3 --datadir=$MYSQLD_DATADIR;
+let $MYSQLD_EXTRA_ARGS = --port=$ro_port --socket=$MYSQLD_DATADIR/test.sock --console --skip-ssl --log-bin=master-bin --objectstore-region=.local_region --objectstore_bucket=$self_bucket --log-error-verbosity=3 --datadir=$MYSQLD_DATADIR;
 --exec $MYSQLD --no-defaults $MYSQLD_ARGS $MYSQLD_EXTRA_ARGS $MYSQLD_EXTRA_INIT_ARGS --initialize-insecure > $MYSQLD_ERROR_LOG 2>&1
 --echo # Start ro server
 let $ro_cluster_info= '127.0.0.1:19981@1';
@@ -119,10 +120,10 @@ let MYSQLD_INNODB_WAL_DIR = $MYSQLD_LOG_DIR;
 let MYSQLD_SE_SST_DATADIR = $MYSQLD_LOG_DIR/smartengine_sst;
 let MYSQLD_SE_WAL_DIR = $MYSQLD_LOG_DIR/smartengine_wal;
 
-let MYSQLD_EXTRA_INIT_ARGS = --initialize_from_objectstore=true --initialize_smartengine_objectstore_data=true --initialize_objectstore_region=$source_region --initialize_objectstore_bucket=$source_bucket --initialize_repo_objectstore_id=$source_repo;
+let MYSQLD_EXTRA_INIT_ARGS = --initialize_from_objectstore=true --initialize_smartengine_objectstore_data=true --initialize_objectstore_region=$source_region --initialize_objectstore_bucket=$source_bucket --initialize_repo_objectstore_id=$source_repo --objectstore_bucket=$self_bucket;
 
 --echo # Initialize ro2 server
-let $MYSQLD_EXTRA_ARGS = --port=$ro2_port --socket=$MYSQLD_DATADIR/test.sock --console --skip-ssl --log-bin=$MYSQLD_BINLOG_DIR --objectstore-region=.local_region --log-error-verbosity=3 --datadir=$MYSQLD_DATADIR --innodb_log_group_home_dir=$MYSQLD_INNODB_WAL_DIR --smartengine_data_dir=$MYSQLD_SE_SST_DATADIR --smartengine_wal_dir=$MYSQLD_SE_WAL_DIR;
+let $MYSQLD_EXTRA_ARGS = --port=$ro2_port --socket=$MYSQLD_DATADIR/test.sock --console --skip-ssl --log-bin=$MYSQLD_BINLOG_DIR --objectstore-region=.local_region --objectstore_bucket=$self_bucket --log-error-verbosity=3 --datadir=$MYSQLD_DATADIR --innodb_log_group_home_dir=$MYSQLD_INNODB_WAL_DIR --smartengine_data_dir=$MYSQLD_SE_SST_DATADIR --smartengine_wal_dir=$MYSQLD_SE_WAL_DIR;
 --exec $MYSQLD --initialize-insecure --no-defaults $MYSQLD_ARGS $MYSQLD_EXTRA_ARGS $MYSQLD_EXTRA_INIT_ARGS > $MYSQLD_ERROR_LOG 2>&1
 --echo # Start ro2 server
 let $ro2_cluster_info= '127.0.0.1:19989@1';

--- a/sql/consistent_recovery.cc
+++ b/sql/consistent_recovery.cc
@@ -1487,7 +1487,7 @@ bool Consistent_recovery::recovery_smartengine_objectstore_data() {
     destination_object_key.append(object_key_suffix);
 
     // push object to destination object store.
-    ss = init_destination_objstore->put_object(opt_objstore_bucket,
+    ss = init_destination_objstore->put_object(m_init_destination_objstore_bucket,
                                                destination_object_key, data);
     if (!ss.is_succ()) {
       err_msg.assign("put object to destination object store failed: ");
@@ -1692,7 +1692,7 @@ int Consistent_recovery::fetch_last_persistent_snapshot_index_file(
    snapshot.index, should locate the last one in list. If only the
    single-version file (snapshot.index) exists, it is selected instead..
   */
-  if (!list_persistent_objects(opt_objstore_bucket, objects,
+  if (!list_persistent_objects(m_objstore_bucket, objects,
                                index_prefix.c_str(), false, true, true)) {
     return 1;
   }
@@ -1807,7 +1807,7 @@ int Consistent_recovery::get_last_persistent_binlog_consensus_index() {
   remove_file(binlog_index_file_name);
   {
     objstore::Status ss = recovery_objstore->get_object_to_file(
-        std::string_view(opt_objstore_bucket), last_index_keyid,
+        std::string_view(m_objstore_bucket), last_index_keyid,
         std::string_view(binlog_index_file_name));
     if (!ss.is_succ()) {
       err_msg.assign("download persistent binlog index file failed: ");

